### ISSUE TITLE
extract_residuals: recognize division by a positive constant (sum_of_squares / c) for Gauss-Newton

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/discopt#100]
## Summary

`extract_residuals` (the Gauss-Newton sum-of-squares detector behind `Model.solve(gauss_newton=True)`, #98) does not recognize **division by a positive constant**. An objective written as `sum_of_squares / c` returns `None` (GN silently falls back to the dense Hessian), even though it is exactly equivalent to `(1/c) * sum_of_squares`, which the matcher *does* recognize via its `c * g` form.

This bites real models: a least-squares objective weighted by a known variance is naturally written `(residual)**2 / sigma` or `dm.sum(R*R) / sigma`. Today that has to be hand-rewritten as `(1.0/sigma) * (...)` for `gauss_newton=True` to take effect (the failure is silent — `is_gauss_newton` just stays `False`).

## Repro

```python
import numpy as np, discopt.modeling as dm
from discopt._jax.least_squares import extract_residuals

m = dm.Model('t'); x = m.continuous('x', shape=(2,))
div = (x[0] - 1.0)**2 / 1e-3 + (x[1] - 2.0)**2 / 1e-3   # natural form
mul = (1.0/1e-3) * (x[0] - 1.0)**2 + (1.0/1e-3) * (x[1] - 2.0)**2

print(extract_residuals(div) is not None)   # False  <-- not recognized
print(extract_residuals(mul) is not None)   # True
```

Same story for the array form: `dm.sum(R*R) / sigma` → `None`, `(1.0/sigma) * dm.sum(R*R)` → recognized.

## Proposed fix

In `extract_residuals`, handle `BinaryOp(op='/')` where the **right operand is a positive scalar constant**: normalize `g / c` → treat as `(1/c) * g` and recurse with the existing `_scale_residuals(inner, sqrt(1/c))` path. Mirror the guard already used for `c * g`:

* right operand must be a scalar constant `c > 0` (reject non-constant denominators and `c <= 0`, exactly as the multiplication path rejects `c < 0`);
* division by a non-constant / array denominator stays unsupported (it does not preserve the sum-of-squares structure — the result can go negative and `2 JᵀJ` no longer holds), so only the positive-scalar-constant case is added.

That makes the natural `... / sigma` weighting work without callers hand-converting to `* (1/sigma)`.

## Scope note

This is purely the constant-denominator case. General `a / b` with `b` an expression is correctly left unrecognized.

(Context: surfaced from kipet2, where every least-squares objective is variance-weighted as `(...)/sigma`; worked around there by multiplying instead, but the natural form silently disabling GN is an easy trap.)
